pennylane_lightning/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -0,0 +1,1678 @@
+// Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file StateVectorKokkos.hpp
+ */
+
+#pragma once
+#include <cstdlib>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+#include "BitUtil.hpp" // isPerfectPowerOf2
+#include "Error.hpp"
+#include "GateFunctors.hpp"
+#include "StateVectorBase.hpp"
+#include "Util.hpp"
+
+/// @cond DEV
+namespace {
+using Pennylane::Util::exp2;
+using Pennylane::Util::isPerfectPowerOf2;
+using Pennylane::Util::log2;
+using namespace Pennylane::LightningKokkos::Functors;
+} // namespace
+/// @endcond
+
+namespace Pennylane::LightningKokkos {
+
+/**
+ * @brief  Kokkos state vector class
+ *
+ * @tparam fp_t Floating-point precision type.
+ */
+template <class fp_t = double>
+class StateVectorKokkos final
+    : public StateVectorBase<fp_t, StateVectorKokkos<fp_t>> {
+
+  private:
+    using BaseType = StateVectorBase<fp_t, StateVectorKokkos<fp_t>>;
+
+  public:
+    using PrecisionT = fp_t;
+    using ComplexT = Kokkos::complex<fp_t>;
+    using KokkosExecSpace = Kokkos::DefaultExecutionSpace;
+    using KokkosVector = Kokkos::View<ComplexT *>;
+    using KokkosSizeTVector = Kokkos::View<size_t *>;
+    using KokkosRangePolicy = Kokkos::RangePolicy<KokkosExecSpace>;
+    using UnmanagedComplexHostView =
+        Kokkos::View<ComplexT *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+    using UnmanagedSizeTHostView =
+        Kokkos::View<size_t *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+    using UnmanagedConstComplexHostView =
+        Kokkos::View<const ComplexT *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+    using UnmanagedConstSizeTHostView =
+        Kokkos::View<const size_t *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+
+    StateVectorKokkos() = delete;
+    StateVectorKokkos(size_t num_qubits,
+                      const Kokkos::InitializationSettings &kokkos_args = {})
+        : BaseType{num_qubits},
+          gates_{// Identity
```

Comment:
Do we have any special to keep the `// Identity`? If not I propose to remove it.

Reply:
I think it was just to state that there is no implementation of `Identity`. I'll remove it.